### PR TITLE
SMB Support - Add new Java DSL components to XSD config

### PR DESCRIPTION
**Pull request by GregBragg**
1. Update XSD config to support Java DSL components
2. Add XML config examples for Java DSL components in SMB documentation

@artembilan This PR addresses previous feedback around providing XML config support for the new components introduced during the effort to add Java DSL support.

### CODE REVIEW

**Review by artembilan — CHANGES_REQUESTED** [review]
Thanks for the effort, but this is not going to work as is.
There is just no void magic in between that xml config and runtime application context.
We need to implement `BeanDefinitionParser`s and register them with the namespace handler to really introduce that magic.

See for example the same `SftpStreamingInboundChannelAdapterParser` and `SftpOutboundGatewayParser` and how they are used in the `SftpNamespaceHandler`.

The next step would be to have some simple unit tests to verify some simple xml configuration for these new tags.

Does it make sense?

**Comment by GregBragg** [comment]
Yes makes sense, I'll look into your suggestions. I admit this part of Spring Integration I'm not entirely experienced with so I understand your concerns.

Thanks for the feedback, hold off on this for now and I will spend some more time next week to review the SFTP code suggestions and to come up with some unit tests to verify functionality. Thanks!

**Review by artembilan — CHANGES_REQUESTED** [review]
Just minor concern.

Let me know when you done with all the changes and I'll pull it locally for final review and merge!

File: spring-integration-smb/src/main/java/org/springframework/integration/smb/config/SmbOutboundGatewayParser.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+import org.springframework.integration.file.config.AbstractRemoteFileOutboundGatewayParser;
+import org.springframework.integration.file.remote.RemoteFileOperations;
+import org.springframework.integration.smb.filters.SmbRegexPatternFileListFilter;
+import org.springframework.integration.smb.filters.SmbSimplePatternFileListFilter;
+import org.springframework.integration.smb.outbound.SmbOutboundGateway;
+import org.springframework.integration.smb.session.SmbRemoteFileTemplate;
+
+/**
+ * @author Gregory Bragg
```
[the remark this answers is not in the document]
Comment (GregBragg) [reply spring-integration-smb/src/main/java/org/springframework/integration/smb/config/SmbOutboundGatewayParser.java:29]: Looks like I am done for now. Let me know of any other concerns... Thanks!

**Pull request closed without merging**